fdbserver/include/fdbserver/IPager.h

```diff
@@ -387,23 +388,40 @@ class ArenaPage : public ReferenceCounted<ArenaPage>, public FastAllocated<Arena
 	// By default, xxhash is used to checksum the page. But ff authentication is enabled (such as when we are using
 	// aes256-ctr-hmac-sha256 encryption scheme), the auth tag plays the role of a checksum while assuring authenticity
 	// of the data. xxhash checksum is not needed in this case.
+	//
+	// To support configurable encryption, which may come with variable size encryption header, we assume the encryption
+	// header size is no larger than that of BlobCipherEncryptHeader. This is true for current supported encryption
+	// header format types. Moving forward, the plan is to make IPager support variable size encoding header, and let
+	// Redwood rebuild a page when it tries to in-place update the page, but the reserved buffer for the encoding header
+	// is not large enough.
+	// TODO(yiwu): Cleanup the old encryption header, and update headerSize to be the maximum size of the supported
+	// encryption header format type.
+	// TODO(yiwu): Support variable size encoding header.
 	template <EncodingType encodingType,
 	          typename std::enable_if<encodingType == AESEncryption || encodingType == AESEncryptionWithAuth,
 	                                  bool>::type = true>
 	struct AESEncryptionEncoder {
 		struct AESEncryptionEncodingHeader {
-			BlobCipherEncryptHeader encryption;
 			XXH64_hash_t checksum;
+			union {
+				BlobCipherEncryptHeader encryption;
+				uint8_t encryptionHeaderBuf[0]; // for configurable encryption
+			};
 		};
 
 		struct AESEncryptionWithAuthEncodingHeader {
-			BlobCipherEncryptHeader encryption;
+			union {
+				BlobCipherEncryptHeader encryption;
```

Comment:
for better understanding: don't we want to add `BlobCipherEncryptHeaderRef` as an element to the union`?

Reply:
This struct holds the serialized encoding header. When configurable encryption is enabled, the `encryptionHeaderBuf` pointer in this union points to the serialized string ref of `BlobCipherEncryptHeaderRef`. The `BlobCipherEncryptHeaderRef` itself represents un/de-serialized header.